src/drive/web/modules/views/Folder/syncHelpers.js

```diff
@@ -0,0 +1,146 @@
+ * @param {string} params.sharingId - Id of an io.cozy.sharings doc
+ */
+export const removeSharingFromContext = ({
+  sharingsValue,
+  setSharingsValue,
+  sharingId
+}) => {
+  delete sharingsValue[sharingId]
+  setSharingsValue(sharingsValue)
+}
+
+/**
+ * Create a syncing fake file
+ * @param {object} params - Params
+ * @param {string} params.sharingId - Id of an io.cozy.sharings doc
+ * @param {object} params.sharingsValue - Sharing Context value
+ * @param {object} params.fileValue - Sharing Context file value
+ * @returns {object} Syncing fake file
+ */
+export const createSyncingFakeFile = ({
+  sharingId,
+  sharingsValue,
+  fileValue
+}) => {
+  if (fileValue) {
+    return null
+  }
+
+  const sharingValue = sharingsValue[sharingId]
+
+  if (!sharingValue) {
+    return null
+  }
+
+  return {
+    name: sharingValue.attributes.description,
+    id: sharingValue.id,
+    type: 'directory'
+  }
+}
+
+/**
+ * Returns syncingFakeFile if it is still needed, otherwise remove the share in context
+ * @param {object} params - Params
+ * @param {array} params.queryResults - List of folders and files
+ * @param {string} params.sharingId - Id of an io.cozy.sharings doc
+ * @param {object} params.sharingsValue - Sharing Context value
+ * @param {function} params.setSharingsValue - Sharing Context setter
+ * @param {object} params.syncingFakeFile - Syncing fake file
+ * @returns {object} Syncing fake file
+ */
+export const checkSyncingFakeFileObsolescence = ({
+  queryResults,
+  sharingId,
+  sharingsValue,
+  setSharingsValue,
+  syncingFakeFile
+}) => {
+  const isThereRealtimeFileReferencedBySharing = isThereFileReferencedBySharingId(
+    queryResults,
+    sharingId
+  )
+
+  if (!isThereRealtimeFileReferencedBySharing) {
+    return syncingFakeFile
+  }
+
+  removeSharingFromContext({ sharingsValue, setSharingsValue, sharingId })
+  return null
+}
+
+/**
+ * Create syncing fake file or check if it still longer needed
+ * @param {object} params - Params
+ * @param {boolean} params.isEmpty - Whether the query to fetch files returns nothing or error
+ * @param {boolean} params.isSharingContextEmpty - Whether the sharing context is empty
+ * @param {array} params.queryResults - List of folders and files
+ * @param {string} params.sharingId - Id of an io.cozy.sharings doc
+ * @param {object} params.sharingsValue - Sharing Context value
+ * @param {function} params.setSharingsValue - Sharing Context setter
+ * @param {object} params.fileValue - Sharing Context file value
+ * @returns {object} Syncing fake file
+ */
+export const computeSyncingFakeFile = ({
+  isEmpty,
+  isSharingContextEmpty,
+  queryResults,
+  sharingId,
+  sharingsValue,
+  setSharingsValue,
+  fileValue
+}) => {
+  if (isEmpty || isSharingContextEmpty) {
+    return null
+  }
+
+  const syncingFakeFile = createSyncingFakeFile({
+    sharingId,
+    sharingsValue,
+    fileValue
+  })
+
+  if (!syncingFakeFile) {
```

Comment:
We do this check because between the call of `createSyncingFakeFile` and this check we think that context can have changed? Having a comment to explain why this check can be useful

Reply:
Non le context ne peut changer qu'à l'exécution de `checkSyncingFakeFileObsolescence` .

C'est pour voir après un re-rendu, déclenché par exemple par le temps réel, si le fake file est toujours nécessaire. Par contre on re-déclenche la création du fake file `createSyncingFakeFile` ce qui, je pense, n'a pas de raison de donner un résultat différent de la fois précédente... cette partie ne semble pas être très optimisée en effet

Reply:
Modification faite, les conditions sont maintenant dans `computeSyncingFakeFile` et non dans `createSyncingFakeFile` qui ne fait que créer le syncingFakeFile